frontend/src/scenes/session-recordings/player/inspector/miniFiltersLogic.ts

```diff
@@ -0,0 +1,303 @@
+
+                    const newFilters = state.filter((existingSelected) => {
+                        const filterInTab = filtersInTab.find((x) => x.key === existingSelected)
+                        if (!filterInTab) {
+                            return true
+                        }
+
+                        if (enabled) {
+                            if (selectedFilter.alone) {
+                                return false
+                            }
+                            return filterInTab.alone ? false : true
+                        }
+
+                        if (existingSelected !== key) {
+                            return true
+                        }
+                        return false
+                    })
+
+                    if (enabled) {
+                        newFilters.push(key)
+                    } else {
+                        // Ensure the first one is checked if no others
+                        if (filtersInTab.every((x) => !newFilters.includes(x.key))) {
+                            newFilters.push(filtersInTab[0].key)
+                        }
+                    }
+
+                    return newFilters
+                },
+            },
+        ],
+
+        searchQuery: [
+            '',
+            {
+                setSearchQuery: (_, { search }) => search || '',
+            },
+        ],
+    })),
+
+    selectors({
+        miniFiltersForTab: [
+            (s) => [s.selectedMiniFilters],
+            (selectedMiniFilters): ((tab: SessionRecordingPlayerTab) => SharedListMiniFilter[]) => {
+                return (tab: SessionRecordingPlayerTab) => {
+                    return MiniFilters.filter((filter) => filter.tab === tab).map((x) => ({
+                        ...x,
+                        enabled: selectedMiniFilters.includes(x.key),
+                    }))
+                }
+            },
+        ],
+
+        miniFilters: [
+            (s) => [s.tab, s.miniFiltersForTab],
+            (tab, miniFiltersForTab): SharedListMiniFilter[] => {
+                return miniFiltersForTab(tab)
+            },
+        ],
+
+        miniFiltersByKey: [
+            (s) => [s.miniFilters],
+            (miniFilters): { [key: string]: SharedListMiniFilter } => {
+                return miniFilters.reduce((acc, filter) => {
+                    acc[filter.key] = filter
+                    return acc
+                }, {})
+            },
+        ],
+
+        miniFiltersForTabByKey: [
+            (s) => [s.miniFiltersForTab],
+            (miniFiltersForTab): ((tab: SessionRecordingPlayerTab) => { [key: string]: SharedListMiniFilter }) => {
+                return (tab) => {
+                    return miniFiltersForTab(tab).reduce((acc, filter) => {
+                        acc[filter.key] = filter
+                        return acc
+                    }, {})
+                }
+            },
+        ],
```

Comment:
the fix here is to have be able to read mini filters for arbitrary tabs
whereas before it was only possible to read the filters for the current tab

the rest is all reorganisation to move things out of playerSettingsLogic which confused me as a home for this stuff